### DISCUSSION TITLE
Laragon Server Index Page

### DISCUSSION THREAD

**Discussion by LebToki** [leokhoa/laragon#447] [Show and tell]
**Welcome to the Laragon root directory index page!**

![screenshot](https://user-images.githubusercontent.com/957618/226374846-86e2ee56-1465-433b-89f5-86f2bf1eac64.png)

Attention developers! Are you tired of sifting through a cluttered directory to find your projects and files? Look no further than our customized user interface designed specifically for your Laragon server.

Our UI automatically detects any projects and directories within your root directory, making it easy to manage your work. Plus, we've included an automated auto-detection for popular frameworks like WordPress, Laravel, and Symfony.

We've gone the extra mile to provide a visually appealing and intuitive interface that's easy to navigate, so you can spend more time coding and less time searching for files. And with a responsive design, you can access our UI on any device.

To use our index page, simply replace the default index.php file in your Laragon root directory with our customized index.php file. And don't worry, we've taken care of cleaning up all PHP calls, providing error responses, and removing redundant codes.

This is good enough to be the official index page of Lragon to replace the vanilla Laragon server index page, and offers several key features that make it an essential tool for developers:

***Visually appealing interface:** Our index page provides a clean and modern design that is both easy on the eyes and easy to navigate.
***Intuitive project management:** With our UI, you can quickly navigate through your projects and directories, making it easy to find the files you need and manage your server more effectively.
***Responsive design:** Our index page is designed to work seamlessly on any device, from desktop computers to mobile phones.

**How to Use**
1. To use our index page, simply replace the default index.php file in your Laragon root directory with our customized index.php file. Once you've done that, you'll be able to access our UI by navigating to the root directory of your Laragon server.
2. Make sure you change your development path (.local) - if you've named your path differently
3. Make sure to upload the icons and css files to the assets directory (or simply import it as is)
4. Make sure to changetheignored directories or add the directories that you don't intend to expose publicly

**STAR THE PROJECT TO BE NOTIFIED IN THE FUTURE AND DO COME BACK WITH YOUR FEEDBACK**
So what are you waiting for? Head over to our GitHub repo to download our customized index page and enhance your development experience. Don't forget to star the project to stay up-to-date on future improvements, and please share your feedback with us. We're always looking for ways to make our index page even better for you.

Go ahead and get your copy!
[from my GitHub Repo](https://github.com/LebToki/Index-for-Laragon)

**Comment by RobiNN1** [comment]


**Reply by LebToki** [reply]
Let me see what can be done... 
why not 👍

**Reply by LebToki** [reply]
I'm having troubles accessing php latest info !
maybe you guys can help resolve the issue so we can merge it into the next version.

[php version is hardcoded issue](https://github.com/LebToki/Index-for-Laragon/issues/1)

**Comment by LebToki** [comment]
Personally i tested it successfully and can guarantee it works except that it will require probably an ajax call and creation of a second file called scan.php that will scan and fetch the directories their favicon and then closes the animation (if we decide to use one anyway)
frankly it's a bit of a hussle. if you are keen to having it then i have a file that is working but i personally won't go there until i find a better solution :)

Here's the updated function, the downside of using it is that it does take some time and the user might think that it is stalling or an error is generated and the page went into a loop (depends on the number of folders obviously)

[dynamic favicon](https://github.com/LebToki/Index-for-Laragon/issues/3)

**Reply by LebToki** [reply]
This is fixed now and merged into v1.2

The hostname is also hardcoded, but there is a HostnameFormat in laragon.ini, so this can also be retrieved automatically. And seem that it doesn't obtains current ServerName
[servername is missing](https://github.com/LebToki/Index-for-Laragon/issues/4)

**Comment by kstingel** [comment]
just a suggestion: more for the next iteration of Laragon, since it would mean tweaking the application's `default index.php`

rename your 'index.php' to `dashboard.php` and (your) entire project could just be unpacked into `'www'` directory
... and possibly rename the project to `Laragon Dashboard`?

**Reply by LebToki** [reply]
I agree but would be much easier to simply replace the existing index.php since this has more features and functionalities !

**Reply by LebToki** [reply]
I'm working on a version that would apply a bit from your direction and therefore move additional files to the same assets directory that is being created
@kstingel 
@luisAntonioLAGS 
I'm introducing what I've promised so would love for you to maintain the Spanish es.json and this way it will save you from hardcoding changes inside the file

**Comment by LebToki** [comment]
I added an Inbox and brushed up the codebase further
[https://github.com/LebToki/Index-for-Laragon](https://github.com/LebToki/Index-for-Laragon)
download your 2.2.1 package now 

![Screenshot 2024-05-27 192319](https://github.com/leokhoa/laragon/assets/957618/4e8ca586-0f98-4428-ada2-2b6f5b09f548)

**Comment by luisAntonioLAGS** [comment]
I modified it so that it is in Spanish. Good job creating this dashboard.
[index_laragon_es.zip](https://github.com/leokhoa/laragon/files/15486936/index_laragon_es.zip)

**Reply by LebToki** [reply]
Nice Thanks for the Efforts Luis 
we can look at some language files, maybe in the future, to save you the headaches of chasing HTML tags

I've tagged you here 
[https://github.com/LebToki/Index-for-Laragon/discussions/16](https://github.com/LebToki/Index-for-Laragon/discussions/16)

**Reply by luisAntonioLAGS** [reply]
> Nice Thanks for the Efforts Luis we can look at some language files, maybe in the future, to save you the headaches of chasing HTML tags
> 
> I've tagged you here [LebToki/Index-for-Laragon#16](https://github.com/LebToki/Index-for-Laragon/discussions/16)

You're welcome, I'm happy to collaborate with software as powerful and free as Laragon. I liked the dashboard you created. Thanks for this contribution to laragon.

Note: Laragon did need a new prettier home page, just as you created it.

**Reply by LebToki** [reply]
next version will have /assets/languages/en.json (multiple languages) es.json
maintainers will be welcomed and praised too

**Reply by LebToki** [reply]
![image](https://github.com/leokhoa/laragon/assets/957618/4533b308-7d48-4ad7-8530-9f549b9df81d)
@leokhoa 
Need your help in providing me these information or exposing them in your new release
Detecting Server Uptime, Memory usage and some other vitals from the whole stack and we can easily plug it in

![image](https://github.com/leokhoa/laragon/assets/957618/d0325037-ba34-42a1-931f-e067b165392a)
Server's Mailbox is now crucial part of the Dashboard and loads the message (allows deleting too) in a modal

![image](https://github.com/leokhoa/laragon/assets/957618/43178c62-b1a7-41f7-ba5a-6f46595c545a)
The Main dashboard now is Multi-Lingual and utilizes Language files

**Reply by luisAntonioLAGS** [reply]
> ![image](https://private-user-images.githubusercontent.com/957618/335598612-4533b308-7d48-4ad7-8530-9f549b9df81d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNzI5MTEsIm5iZiI6MTcxNzE3MjYxMSwicGF0aCI6Ii85NTc2MTgvMzM1NTk4NjEyLTQ1MzNiMzA4LTdkNDgtNGFkNy04NTMwLTlmNTQ5YjlkZjgxZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUzMVQxNjIzMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYTkxZGVjODJiMjNlYjRjYjBiN2UzMDhlNGQwYzI0NTE2Mjc0N2ZlNjNmYjVkNTIxODg2NjRmMGVhZWFjYmQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.AsdcG33nr_BZjqesRrfRI20yTJl-_opzCmNK8u7jd9Q) @leokhoa Need your help in providing me these information or exposing them in your new release Detecting Server Uptime, Memory usage and some other vitals from the whole stack and we can easily plug it in
> 
> ![image](https://private-user-images.githubusercontent.com/957618/335599357-d0325037-ba34-42a1-931f-e067b165392a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNzI5MTEsIm5iZiI6MTcxNzE3MjYxMSwicGF0aCI6Ii85NTc2MTgvMzM1NTk5MzU3LWQwMzI1MDM3LWJhMzQtNDJhMS05MzFmLWUwNjdiMTY1MzkyYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUzMVQxNjIzMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYmU2MDM1N2FiOTAxM2ZhNmQ5ODJlNmNhNzY3OTA0ZTQzMThhNTM2YWE1YzlhMGUxYjI5MTVkNmQwMWVmZDMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kS1QYLbWYJesTXYpF4HYJmU1J3WYpWbF5_IQsv4itx8) Server's Mailbox is now crucial part of the Dashboard and loads the message (allows deleting too) in a modal
> 
> ![image](https://private-user-images.githubusercontent.com/957618/335599677-43178c62-b1a7-41f7-ba5a-6f46595c545a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNzI5MTEsIm5iZiI6MTcxNzE3MjYxMSwicGF0aCI6Ii85NTc2MTgvMzM1NTk5Njc3LTQzMTc4YzYyLWIxYTctNDFmNy1iYTVhLTZmNDY1OTVjNTQ1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUzMVQxNjIzMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMGE4ZWRhMjBlNzQ0NjEwNmJmOGUwM2NkZWM2ZGMxOTQ5MWJmYzRmNTRhNTYxYmMzZmYzZTNkNjRiNjUzZTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1VV7Xzl93qjB6jeIGqAUDbYs7x4rr0WZZ1K43FrpI7U) The Main dashboard now is Multi-Lingual and utilizes Language files

I will do any implementation that I can help you with, lately I have been busy so I missed translating the inbox page, but I would like to be able to support even in the slightest.

**Reply by LebToki** [reply]
Download the 2.3.3 package as i have moved the sub-pages and language files to /assets/
and this way you can start on the right track

**Reply by luisAntonioLAGS** [reply]
> Download the 2.3.3 package as i have moved the sub-pages and language files to /assets/ and this way you can start on the right track

Ok, Thanks!

**Comment by LebToki** [comment]
## 🚀 Key Highlights of the Update v. 2.6.0:

- Enhanced Developer Experience
Framework Detection - Automatic recognition of WordPress, Laravel, Drupal saves time
Quick Access Links - Direct admin panel access boosts productivity
Self-updating Future - No manual updates needed going forward

- Comprehensive Monitoring
Real-time Server Vitals - Live charts for system metrics
Memory & Disk Tracking - Proactive resource management
Performance Metrics - Page load times and execution stats

- Powerful Email Management
Advanced Search & Filter - Find emails efficiently
Bulk Operations - Manage multiple emails easily
Export Capabilities - Download email data for analysis

- Database Tools
Safe Query Execution - Read-only SELECT queries prevent accidents
Table Exploration - Detailed structure and size information
Database Overview - Quick size and status checking

💡 Suggested Installation/Update Instructions:
For users who need to update, you might want to include:

```bash
# Update steps might look like:
cd /path/to/laragon-dashboard
git pull origin main
composer install
```

# Restart Laragon services

🎯 Potential Future Enhancements:
Backup/restore functionality
SSL certificate management
Multi-environment support
Team collaboration features

This update significantly elevates Laragon from a simple local server stack to a full-featured development environment manager. 
The self-updating capability in future versions is particularly smart for maintaining user engagement and promising an early access to updates as they are available.

**Comment by LebToki** [comment]
<img width="1666" height="915" alt="image" src="https://github.com/user-attachments/assets/ed7fb1bc-846f-4104-9004-4cb133de27a9" />
A complete pour of love for the community is coming soon to Laragon Dashboard with more features

**Reply by AWqxKAWERbXo** [reply]
Now this looks much cleaner and visually focused on the actual information and interaction. 

You could actually deep-link IDEs and editors like VSCode etc.  with the current project/folder.

**Reply by LebToki** [reply]
Still cleaning up the whole thing. prior to publishing

**Reply by LebToki** [reply]
<img width="1669" height="917" alt="image" src="https://github.com/user-attachments/assets/57809b2b-7ac7-4cfb-83c9-a74d99562425" />

Looking at a full integration with Laragon / Platforms and Tools and even MailPIT with the API

**Reply by LebToki** [reply]
<img width="1896" height="796" alt="image" src="https://github.com/user-attachments/assets/503443f6-d5bf-4a3d-9f2c-c0ed0269147e" />
Another teaser in refinement

**Reply by LebToki** [reply]
<img width="1650" height="912" alt="image" src="https://github.com/user-attachments/assets/306ab34c-3ba7-44d9-b438-5b9403fafe58" />
How about that ? Exciting ?
There is more love pouring in promise!

**Reply by LebToki** [reply]
<img width="1662" height="910" alt="image" src="https://github.com/user-attachments/assets/3b88c598-7472-42ef-a3d1-da0c2f311831" />

**Reply by LebToki** [reply]
<img width="1881" height="884" alt="image" src="https://github.com/user-attachments/assets/a91ea12e-a195-4a1e-8f58-bbaa064117be" />

**Reply by LebToki** [reply]
<img width="1663" height="916" alt="image" src="https://github.com/user-attachments/assets/37c8fbaf-2f87-45f0-8616-ca88f7d4fa9a" />

**Reply by LebToki** [reply]
<img width="1660" height="914" alt="image" src="https://github.com/user-attachments/assets/951dba39-44eb-4f4c-9f20-9c404e3146e0" />

**Reply by LebToki** [reply]
<img width="1898" height="910" alt="image" src="https://github.com/user-attachments/assets/94abf36b-7e88-4804-8e90-4300dc1d2299" />
